### ISSUE TITLE
Add basic support for ipv6

### ISSUE THREAD

**Issue by StevenJiang1110** [asterinas/asterinas#452]
Gvisor requires basic support for ipv6. That is to say, we should support listen/bind/connect to ipv6 local host address, and support map ipv6 addr to ipv4 if it can.

**Comment by yydawx** [comment]
I'll work on it

**Comment by StevenJiang1110** [comment]
> I'll work on it

Thanks for your interest! Nobody has worked on this yet, so feel free to go ahead!

**Comment by yydawx** [comment]
Adding IPv6 support means changing conditions in a lot of places, so my initial PR/commit will probably be big. Anything I need to be aware of?

Also, I’ve only added regression tests so far. Are there any other specific test cases or locations I should cover based on our usual practices?

Thanks in advance for taking the time to help out!

**Comment by StevenJiang1110** [comment]
> Adding IPv6 support means changing conditions in a lot of places, so my initial PR/commit will probably be big. 

Yes, I understand that full IPv6 support is a substantial undertaking. If possible, I'd encourage you to break the work into several relatively self-contained PRs to help speed up the review process. The first PR should ideally cover only the essential functionality, with optional parts deferred to follow-up PRs. For example, if TCP and UDP support for IPv6 can be implemented independently, it might make sense to submit one before the other — though of course this depends on your judgment, and I'm not sure how the implementation naturally splits.

> Anything I need to be aware of?

It would be worth taking some time to read through the repository's [coding guidelines](https://asterinas.github.io/book/to-contribute/coding-guidelines/index.html), which outline the general expectations for code in this repository, including formatting, naming conventions, and code style. These will also be checked during the code review process, so making sure your submission aligns with them upfront will help significantly reduce the back-and-forth during review.

> Also, I’ve only added regression tests so far. Are there any other specific test cases or locations I should cover based on our usual practices?

You might want to try enabling the gVisor tests. We currently have the IPv6-related tests disabled — you can try removing the following line and see if the IPv6 tests can be enabled. To run the tests, use `make run_kernel AUTO_TEST=conformance CONFORMANCE_TEST_SUITE=gvisor`.

https://github.com/asterinas/asterinas/blob/9e961e07d414a44be3e8ccb82681fc9f79f9a3ba/test/initramfs/src/conformance/gvisor/blocklists/tcp_socket_test#L1

**Comment by yydawx** [comment]
I have submitted the core IPv6 support to PR #3129, which enables basic testing to become possible.

Current Status:

- IPv6 address handling in aster-bigtcp library
- IPv6 socket connection and binding in kernel network layer
- AF_INET6 syscall support (SOCK_STREAM and SOCK_DGRAM)
- IPv6 TCP test cases added
Remaining Work:

- IPv6 UDP implementation is not yet complete (tests marked as skipped)
- Some edge cases in IPv6 address parsing/validation may need refinement
- No IPv6-specific error handling paths tested
Next Steps:

1. Complete IPv6 UDP support
2. Add more comprehensive IPv6 edge case tests
3. IPv6 multicast support (if needed)
Question: Would it be appropriate to set supporting gVisor IPv6 tests as the completion goal for this issue? This would provide a clear target for the remaining work and ensure full compatibility with gVisor's IPv6 test suite.

**Comment by StevenJiang1110** [comment]
> Would it be appropriate to set supporting gVisor IPv6 tests as the completion goal for this issue?

Of course, if the gVisor tests pass, that would be a significant milestone — and we could close this issue.

**Comment by yydawx** [comment]
Hi! I've update PR #3129. Please check and review it when you're available. Thank you~
